cmake/onnxruntime_nblas.cmake

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
```

Comment:
Would be good to have a comment here saying that nblas is used by nuphar, and a description of what it provides

Reply:
Why do we have something called nblas? Could we make it into MLAS? We don't need many confusing names.

Reply:
Also, core/codegen is the home for general-purpose stuff applicable to many backends. So please move anything specific to nuphar to core/providers/nuphar.

Reply:
This is our first version of int8 GEMV implementation and now replaced by tensorization. Will remove to reduce confusion.

---
In reply to: [312440717](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312440717) [](ancestors = 312440717)

Reply:
Yes, that's the design. The nblas lib requires its own cmake because it need to build with AVX2, while we don't want to make Nuphar to be AVX2 specific. Customers might need to run on machines without AVX2. Again, this won't be an issue once we replace nblas with tensorization.

---
In reply to: [312441381](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312441381) [](ancestors = 312441381)

Reply:
renamed to onnxruntime_nuphar_extern

---
In reply to: [312566522](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312566522) [](ancestors = 312566522,312440717)